bin/start_sync_stop.py

```diff
@@ -134,4 +151,45 @@ async def loop(path='rippled', config='rippled.cfg', port=5005):
     level=logging.INFO,
     datefmt='%Y-%m-%d %H:%M:%S',
 )
-asyncio.run(loop())
+
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+parser.add_argument(
+    'rippled',
+    type=Path,
+    nargs='?',
+    default=DEFAULT_EXE,
+    help='Path to rippled.',
+)
+parser.add_argument(
+    '--conf',
+    type=Path,
+    default=DEFAULT_CONFIG,
+    help='Path to configuration file.',
+)
+parser.add_argument(
+    '--port',
+    type=int,
+    default=DEFAULT_PORT,
+    help='The port chosen for JSON RPC.',
+)
+parser.add_argument(
+    '--duration',
+    type=int,
+    default=DEFAULT_SYNC_DURATION,
+    help='Number of contiguous seconds required in a synchronized state.',
+)
+parser.add_argument(
+    '--interval',
+    type=int,
+    default=DEFAULT_POLL_INTERVAL,
+    help='Number of seconds to wait between polls of state.',
+)
+args = parser.parse_args()
+
+
+def test():
+    return sync(args.port, duration=args.duration, interval=args.interval)
+
+
+asyncio.run(loop(test, exe=args.rippled, config=args.conf))
```

Comment:
Specifying an "interesting" path on the command line causes the program to error out.  Here's the command line I specified:
```
python3 start_sync_stop.py ../build/rippled --conf ../../../.config/ripple/rippled.cfg
```
Here's the traceback:
```
Traceback (most recent call last):
  File "start_sync_stop.py", line 195, in <module>
    asyncio.run(loop(test, exe=args.rippled, config=args.conf))
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
    return future.result()
  File "start_sync_stop.py", line 128, in loop
    async with rippled(exe, config) as process:
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/contextlib.py", line 170, in __aenter__
    return await self.gen.__anext__()
  File "start_sync_stop.py", line 42, in rippled
    stderr=subprocess.DEVNULL,
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/subprocess.py", line 217, in create_subprocess_exec
    stderr=stderr, **kwds)
  File "/usr/local/Cellar/python/3.7.7/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/base_events.py", line 1533, in subprocess_exec
    f"program arguments must be a bytes or text string, "
TypeError: program arguments must be a bytes or text string, not PosixPath
```

Reply:
You're using Python 3.7 which doesn't automatically stringify `Path` arguments to `subprocess` functions. I can easily make some changes to accommodate that though, and add an assertion for the minimum required version (right now it's just a comment).